src/waveform/waveformwidgetfactory.cpp

```diff
@@ -430,31 +447,42 @@ int WaveformWidgetFactory::getVSyncType() {
 }
 
 bool WaveformWidgetFactory::setWidgetType(WaveformWidgetType::Type type) {
-    if (type == m_type)
+    return setWidgetType(type, m_type);
+}
+
+bool WaveformWidgetFactory::setWidgetType(WaveformWidgetType::Type type, WaveformWidgetType::Type& currentType) {
+    if (type == currentType)
         return true;
 
     // check if type is acceptable
-    for (int i = 0; i < m_waveformWidgetHandles.size(); i++) {
-        WaveformWidgetAbstractHandle& handle = m_waveformWidgetHandles[i];
-        if (handle.m_type == type) {
-            // type is acceptable
-            m_type = type;
-            if (m_config) {
-                m_config->set(ConfigKey("[Waveform]","WaveformType"), ConfigValue((int)m_type));
-            }
-            return true;
+    int index = findHandleIndexFromType(type);
+    if (index > -1) {
+        // type is acceptable
+        currentType = type;
+        if (m_config) {
+            m_config->set(ConfigKey("[Waveform]", "WaveformType"), ConfigValue((int)(currentType)));
```

Comment:
Not your fault, but we prefer static_cast<int>()
You can also the setValue template.

The same below.